Reproducible - null Query in BoostingQueryBuilderTests throws assertion

Ran into this error while regression testing master.
```
Suite: org.elasticsearch.index.query.BoostingQueryBuilderTests
  2> REPRODUCE WITH: gradle :core:test -Dtests.seed=1817BD8B51E70A2 -Dtests.class=org.elasticsearch.index.query.BoostingQueryBuilderTests -Dtests.method="testToQuery" -Des.logger.level=WARN -Dtests.security.manager=true -Dtests.locale=es_EC -Dtests.timezone=America/Martinique
FAILURE 0.01s J1 | BoostingQueryBuilderTests.testToQuery <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: null
   >      but: was <mapped_boolean:F^0.1>
   > 	at __randomizedtesting.SeedInfo.seed([1817BD8B51E70A2:F67A79E6C49DB548]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.index.query.AbstractQueryTestCase.assertLuceneQuery(AbstractQueryTestCase.java:528)
   > 	at org.elasticsearch.index.query.AbstractQueryTestCase.testToQuery(AbstractQueryTestCase.java:476)
   > 	at java.lang.Thread.run(Thread.java:745)
```

--- a/core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java
+++ b/core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java
@@ -22,6 +22,7 @@ package org.elasticsearch.index.query;
 import com.carrotsearch.randomizedtesting.generators.CodepointSetGenerator;
 import com.fasterxml.jackson.core.JsonParseException;
 import com.fasterxml.jackson.core.io.JsonStringEncoder;
+
 import org.apache.lucene.search.BoostQuery;
 import org.apache.lucene.search.Query;
 import org.apache.lucene.search.TermQuery;
@@ -128,6 +129,7 @@ public abstract class AbstractQueryTestCase<QB extends AbstractQueryBuilder<QB>>
     private static IndicesQueriesRegistry indicesQueriesRegistry;
     private static QueryShardContext queryShardContext;
     private static IndexFieldDataService indexFieldDataService;
+    private static int queryNameId = 0;
 
 
     protected static QueryShardContext queryShardContext() {
@@ -316,10 +318,19 @@ public abstract class AbstractQueryTestCase<QB extends AbstractQueryBuilder<QB>>
                 query.boost(2.0f / randomIntBetween(1, 20));
             }
             if (randomBoolean()) {
-                query.queryName(randomAsciiOfLengthBetween(1, 10));
+                query.queryName(createUniqueRandomName());
             }
         }
         return query;
+    }
+
+    /**
+     * make sure query names are unique by suffixing them with increasing counter
+     */
+    private static String createUniqueRandomName() {
+        String queryName = randomAsciiOfLengthBetween(1, 10) + queryNameId;
+        queryNameId++;
+        return queryName;
     }
 
     /**

